Tools/wasm/emscripten/__main__.py

```diff
@@ -0,0 +1,325 @@
+#!/usr/bin/env python3
+
+import argparse
+import contextlib
+import functools
+import os
+
+try:
+    from os import process_cpu_count as cpu_count
+except ImportError:
+    from os import cpu_count
+from pathlib import Path
+import shutil
+import subprocess
+import sys
+import sysconfig
+import tempfile
+
+WASM_DIR = Path(__file__).parent
```

Comment:
With the move to the emscripten subdir, should this be
```suggestion
WASM_DIR = Path(__file__).parent.parent
```
?